cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -283,18 +283,36 @@ pScriptWitnessFiles witctx autoBalanceExecUnits scriptFlagPrefix scriptFlagPrefi
 
 pScriptDataOrFile :: String -> String -> String -> Parser ScriptDataOrFile
 pScriptDataOrFile dataFlagPrefix helpTextForValue helpTextForFile =
-      ScriptDataFile  <$> pScriptDataFile
-  <|> ScriptDataValue <$> pScriptDataValue
+      pScriptDataCborFile
+  <|> pScriptDataJsonFile
+  <|> pScriptDataFile -- TODO Deprecated. Remove this option in the future
+  <|> pScriptDataValue
   where
-    pScriptDataFile =
+    pScriptDataCborFile = ScriptDataCborFile <$>
       Opt.strOption
-        (  Opt.long (dataFlagPrefix ++ "-file")
-        <> Opt.metavar "FILE"
+        (  Opt.long (dataFlagPrefix ++ "-cbor-file")
+        <> Opt.metavar "CBOR FILE"
+        <> Opt.help (helpTextForFile ++ " The file must follow the special \
+                                         \JSON schema for script data.")
+        )
+
+    pScriptDataJsonFile = ScriptDataJsonFile <$>
+      Opt.strOption
+        (  Opt.long (dataFlagPrefix ++ "-json-file")
+        <> Opt.metavar "JSON FILE"
         <> Opt.help (helpTextForFile ++ " The file must follow the special \
                                          \JSON schema for script data.")
         )
 
-    pScriptDataValue =
+    pScriptDataFile = ScriptDataFile (Text.pack ("--" ++ dataFlagPrefix ++ "-file option is deprecated")) <$>
```

Comment:
Why is this being deprecated?

Reply:
I think `--tx-in-redeemer-json-file` is clearer than `--tx-in-redeemer-file`, although I'm not sure if our deprecation policy allows this.

If we prefer to keep the original option name then I can revert that portion.

Reply:
I'd say keep the original option name

Reply:
Done!